src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_nd_kernel_ref.cpp

```diff
@@ -155,9 +156,32 @@ bool GatherNDKernelRef::Validate(const Params& p, const optional_params& o) cons
         return false;
     }
 
-    for (uint8_t i = 0; i < batch_dims; i++) {
-        if (input_dims[i] != indices_dims[i]) {
+    if (!params.inputs[0].is_dynamic()) {
+        for (uint8_t i = 0; i < batch_dims; i++) {
+            if (input_dims[i] != indices_dims[i]) {
+                return false;
+            }
+        }
+    }
+
+    if (params.outputs[0].is_dynamic()) {
+        auto supported_tensor_layout = [](const DataTensor& t) -> bool {
+            if (t.GetLayout() == DataLayout::bfyx ||
+                t.GetLayout() == DataLayout::bfzyx ||
+                t.GetLayout() == DataLayout::bfwzyx) {
```

Comment:
Do we need this check here? It is already checked in impelementation_map

Reply:
Agreed and Fixed. Thanks!